webui/react/src/services/apiBuilder.ts

```diff
@@ -54,3 +59,22 @@ export function generateApi<Input, Output>(api: Api<Input, Output>) {
     }
   };
 }
+
+export const consumeStream = async <T = unknown>(
+  fetchArgs: DetSwagger.FetchArgs,
+  onEvent: (event: T) => void,
+  onFinish: () => void): Promise<void> => {
+  let response;
+  try {
+    response = await fetch(serverAddress() + fetchArgs.url, fetchArgs.options);
+  } catch (e) {
+    return processApiError(fetchArgs.url, e);
+  }
+  const exampleReader = ndjsonStream(response.body).getReader();
+  let result;
+  while (!result || !result.done) {
+    result = await exampleReader.read();
+    if (result.done) return onFinish();
+    onEvent(result.value.result);
+  }
```

Comment:
suggestion: move everything after the `catch` block into the `try` block to capture any unforeseen errors to exit gracefully via `processApiError`. Otherwise it may halt React.

Reply:
good idea, should we still include that as an "API" error?